### ISSUE TITLE
Data for Worker overload

### ISSUE THREAD

**Issue by bmarty** [vector-im/element-android#2721]
Serialized data we pass to the worker are limited in size.

On crowded encrypted rooms it can lead to this error:

```
E/SyncThread: java.lang.IllegalStateException: Data cannot occupy more than 10240 bytes when serialized
        at androidx.work.Data.toByteArrayInternal(Data.java:417)
        at androidx.work.Data$Builder.build(Data.java:846)
        at org.matrix.android.sdk.internal.worker.WorkerParamsFactory.toData(WorkerParamsFactory.kt:44)
        at org.matrix.android.sdk.internal.crypto.crosssigning.DefaultCrossSigningService.onUsersDeviceUpdate(DefaultCrossSigningService.kt:809)
        at org.matrix.android.sdk.internal.session.room.summary.RoomSummaryUpdater.update(RoomSummaryUpdater.kt:150)
        at org.matrix.android.sdk.internal.session.room.summary.RoomSummaryUpdater.update$default(RoomSummaryUpdater.kt:67)
        at org.matrix.android.sdk.internal.session.sync.RoomSyncHandler.handleJoinedRoom(RoomSyncHandler.kt:189)
        at org.matrix.android.sdk.internal.session.sync.RoomSyncHandler.handleRoomSync(RoomSyncHandler.kt:110)
        at org.matrix.android.sdk.internal.session.sync.RoomSyncHandler.handle(RoomSyncHandler.kt:93)
```

This part has to be reworked to avoid this overflow.

P1: Once this error occurs, the sync thread is broken, so new messages does not arrive, and only a clear cache seems to fix the problem

**Comment by bmarty** [comment]
https://github.com/matrix-org/element-android-rageshakes/issues/18859

**Comment by bmarty** [comment]
Fixed in Element 1.0.15
